TrainingLoop/Callbacks/StatisticsRecorder.swift

```diff
@@ -121,3 +121,34 @@ public class StatisticsRecorder {
     return result
   }
 }
+
+extension StatisticsRecorder {
+  /// The events on which statistics will be reported for training and validation phases.
+  public enum ReportTrigger {
+    /// Report statistics at end of training and validation, once per epoch.
+    case endOfEpoch
+    /// Report statistics at end of training and validation, once per batch.
+    case endOfBatch
+  }
+
+  /// Set the StatisticsRecorder to report statistics when `trigger` is triggered.
```

Comment:
Grammatically, it would be “Sets…”, not “Set…”. Do you understand the difference?  It's called the “indicative mood.”

Triggers don't get triggered; they trigger something else (e.g. a gun firing).

```suggestion
  /// Updates `self` to report statistics when `trigger` is encountered.
```

Reply:
Learnt it. :)